blocks/hooks/deprecated.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { includes } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, DangerousHTML } from '@wordpress/element';
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Wrapper component for DangerousHTML, logging a warning about unsupported raw
+ * markup return values from a block's `save` implementation.
+ */
+export class DangerousHTMLWithWarning extends Component {
+	constructor() {
+		super( ...arguments );
+
+		// Disable reason: We're intentionally logging a console warning
+		// advising the developer to upgrade usage.
+
+		// eslint-disable-next-line no-console
+		console.warn(
+			'Deprecated: Returning raw HTML from block `save` is not supported. ' +
+			'Use `wp.element.DangerousHTML` component instead.\n\n' +
+			'See: https://wordpress.org/gutenberg/handbook/block-api/block-edit-save/#save'
+		);
+	}
+
+	render() {
+		const { children } = this.props;
+
+		return <DangerousHTML>{ children }</DangerousHTML>;
+	}
+}
+
+/**
+ * Override save element for a block, providing support for deprecated HTML
+ * return value, logging a warning advising the developer to use the preferred
+ * DangerousHTML component instead.
+ *
+ * @param {WPElement} element Original block save return.
+ *
+ * @returns {WPElement}       Dangerously shimmed block save.
+ */
+export function shimDangerousHTML( element ) {
+	// Still support string return from save, but in the same way any component
+	// could render a string, it should be escaped. Therefore, only shim usage
+	// which had included some HTML expected to be unescaped.
+	if ( typeof element === 'string' && includes( element, '<' ) ) {
```

Comment:
`includes( element, '<' )` - this behavior might lead to some errors. I will comment next to the code which might cause issues.

Reply:
>`includes( element, '<' )` - this behavior might lead to some errors. I will comment next to the code which might cause issues.

Which comment expanded on the issue here?

Reply:
https://github.com/WordPress/gutenberg/pull/4786#discussion_r165915137

Reply:
Talked with @gziolo about this some more. Basically: We want to support string returns from `save`, but in the same way that string is a valid React element, it should be unescaped. The shim here specifically targets HTML which previous block implementations may have been returning. The one case where this won't work well is if a block had returned a shortcode string, which will now be escaped. We could similarly add support for shortcode patterns, or alternatively just allow them to break (it's unclear to me how common this would have been).